tensorpipe/CMakeLists.txt

```diff
@@ -4,8 +4,16 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-add_library(tensorpipe
-  ${TP_STATIC_OR_SHARED}
+# List of source files that we need to build libtensorpipe.
+set(TP_PUBLIC_SRCS)
+set(TP_PRIVATE_SRCS)
+set(TP_CUDA_PRIVATE_SRCS)
```

Comment:
As I mentioned in the other PR, I don't see the reason to distinguish public vs private sources. They both end up in the same library. Are there downsides to merging them and just using two variables (`TP_SRCS` and `TP_CUDA_SRCS`)? (Note that headers would still need to be separated between public and private)

Reply:
I have updated the changes. Was thinking of any symbol conflicts which would arise with PUBLIC.